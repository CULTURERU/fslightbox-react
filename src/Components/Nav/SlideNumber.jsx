import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { ifOnUpdateExistsHandleItForState } from "../../Helpers/State/ifOnUpdateExistsHandleItForState";

const SlideNumber = (
    {
        fsLightbox: {
            data: { totalSlides },
            componentsStates: { slide: slideState },
        }
    }
) => {
    const [slide, setSlide] = useState(1);
    slideState.get = () => slide;
    slideState.set = setSlide;

    useEffect(() => {
        ifOnUpdateExistsHandleItForState(slideState);
    });

    if (totalSlides === 1) {
        return null;
    }

    return (
        <div className="fslightbox-slide-number-container fslightbox-flex-centered">
            <div>{ slide }</div>
            <div className="fslightbox-slash">/</div>
            <div>{ totalSlides }</div>
        </div>
    );
};

SlideNumber.propTypes = {
    fsLightbox: PropTypes.object.isRequired
};
export default SlideNumber;