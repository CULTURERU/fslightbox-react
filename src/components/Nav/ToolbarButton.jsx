import React, { Component } from 'react';
import Svg from "../Helpers/Svg.jsx";

class ToolbarButton extends Component {
    render() {
        return (
            <div onClick={ this.props.onClick } className="fslightbox-toolbar-button button-style">
                <Svg viewBox={ this.props.viewBox } size={ this.props.size } d={ this.props.d }/>
            </div>
        );
    }
}

export default ToolbarButton;