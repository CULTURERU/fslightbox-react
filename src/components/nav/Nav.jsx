import React, { Component } from 'react';
import Toolbar from "./Toolbar.jsx";
import SlideNumber from "./SlideNumber.jsx";
import PropTypes from 'prop-types';

class Nav extends Component {
    render() {
        return (
            <div className="fslightbox-nav">
                <Toolbar closeLightbox={ this.props.closeLightbox }/>
                <SlideNumber
                    slide={ this.props.slide }
                    totalSlides={ this.props.totalSlides }
                />
            </div>
        );
    }
}

Nav.propTypes = {
    slide: PropTypes.number,
    totalSlides: PropTypes.number,
    closeLightbox: PropTypes.func
};
export default Nav;