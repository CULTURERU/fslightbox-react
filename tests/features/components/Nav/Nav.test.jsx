import React from 'react';
import { shallow } from 'enzyme';
import Nav from "../../../../src/components/nav/Nav";

const fsLightbox = {};
const nav = shallow(<Nav fsLightbox={ fsLightbox }/>);

describe('nav DOM', () => {
    it('should match snapshot', () => {
        expect(nav).toMatchSnapshot();
    });
});
