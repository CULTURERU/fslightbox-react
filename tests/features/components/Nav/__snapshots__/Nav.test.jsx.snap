// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav DOM should match snapshot 1`] = `
<div
  className="fslightbox-nav"
>
  <Toolbar
    fsLightbox={Object {}}
  />
  <SlideNumber
    fsLightbox={Object {}}
  />
</div>
`;

exports[`testing DOM should match snapshot 1`] = `
<div
  className="fslightbox-nav"
>
  <Toolbar
    fsLightbox={Object {}}
  />
  <SlideNumber
    fsLightbox={Object {}}
  />
</div>
`;
